PROTO_tests/tests/series_test.py

```diff
@@ -0,0 +1,193 @@
+        assert val_sort.index.to_pandas().tolist() == perm.to_list()
+        assert val_sort.values.to_list() == ordered.to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", DTYPES)
+    def test_head_tail(self, prob_size, dtype):
+        s = ak.Series(ak.arange(prob_size, dtype=dtype))
+
+        for i in range(prob_size):
+            head = s.head(i)
+            assert head.index.to_list() == list(range(i))
+            assert head.values.to_list() == ak.arange(i, dtype=dtype).to_list()
+
+            tail = s.tail(i)
+            assert tail.index.to_list() == ak.arange(prob_size)[-i:prob_size].to_list()
+            assert tail.values.to_list() == ak.arange(prob_size, dtype=dtype)[-i:prob_size].to_list()
+
+    def test_value_counts(self):
+        s = ak.Series(ak.array([1, 2, 0, 2, 0]))
+
+        c = s.value_counts()
+        assert c.index.to_list() == [0, 2, 1]
+        assert c.values.to_list() == [2, 2, 1]
+
+        c = s.value_counts(sort=False)
+        assert c.index.to_list() == list(range(3))
+        assert c.values.to_list() == [2, 1, 2]
+
+    def test_concat(self):
+        s = ak.Series(ak.arange(5))
+        s2 = ak.Series(ak.arange(5, 11), ak.arange(5, 11))
+        s3 = ak.Series(ak.arange(5, 10), ak.arange(5, 10))
+
+        df = ak.Series.concat([s, s2], axis=1)
+        assert isinstance(df, ak.DataFrame)
+
+        ref_df = pd.DataFrame(
+            {
+                "idx": list(range(11)),
+                "val_0": [0, 1, 2, 3, 4, 0, 0, 0, 0, 0, 0],
+                "val_1": [0, 0, 0, 0, 0, 5, 6, 7, 8, 9, 10],
+            }
+        )
+        assert_frame_equal(ref_df, df.to_pandas())
+
+        def list_helper(arr):
+            return arr.to_list() if isinstance(arr, (ak.pdarray, ak.Index)) else arr.tolist()
+
+        for fname in "concat", "pdconcat":
+            func = getattr(ak.Series, fname)
+            c = func([s, s2])
+            assert list_helper(c.index) == list(range(11))
+            assert list_helper(c.values) == list(range(11))
+
+            df = func([s, s3], axis=1)
+            if fname == "concat":
+                ref_df = pd.DataFrame(
+                    {"idx": [0, 1, 2, 3, 4], "val_0": [0, 1, 2, 3, 4], "val_1": [5, 6, 7, 8, 9]}
+                )
+                assert isinstance(df, ak.DataFrame)
+                assert_frame_equal(ref_df, df.to_pandas())
+            else:
+                ref_df = pd.DataFrame({0: [0, 1, 2, 3, 4], 1: [5, 6, 7, 8, 9]})
+                assert isinstance(df, pd.DataFrame)
+                assert_frame_equal(ref_df, df)
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_index_as_index_compat(self, prob_size):
+        # added to validate functionality for issue #1506
+        df = ak.DataFrame(
+            {"a": ak.arange(prob_size), "b": ak.arange(prob_size), "c": ak.arange(prob_size)}
+        )
+        g = df.groupby(["a", "b"])
```

Comment:
we aren't gaining anything by testing prob_size here

Reply:
Updated